src/Vocento/Composer/Installers/GitIgnore.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/*
+* This file is part of the Vocento Software.
+*
+* (c) Vocento S.A., <desarrollo.dts@vocento.com>
+*
+* For the full copyright and license information, please view the LICENSE
+* file that was distributed with this source code.
+*
+*/
+
+namespace Vocento\Composer\Installers;
+
+/**
+ * @author  Emilio Fernández <efernandez@vocento.com>
+ * @author  Jorge Marquez <jmarquezz@paradigmadigital.com>
+ */
+class GitIgnore
+{
+    /**
+     * @var array
+     */
+    protected $lines = array();
+    /**
+     * @var string|null
+     */
+    protected $gitIgnoreLocation;
+    /**
+     * @var bool
+     */
+    protected $hasChanges = false;
+    
+    /**
+     * @param string $fileLocation
+     */
+    public function __construct($fileLocation, $filesystem)
+    {
+        $this->gitIgnoreLocation = $fileLocation.DIRECTORY_SEPARATOR.'.gitignore';
+        if (file_exists($this->gitIgnoreLocation)) {
+            $this->lines = $this->removeDuplicates(file($this->gitIgnoreLocation, FILE_IGNORE_NEW_LINES));
+        } else {
+            $filesystem->touch($this->gitIgnoreLocation);
+        }
+    }
+    
+    /**
+     * @param string $file
+     */
+    public function addEntry($file)
+    {
+        $file = $this->prependSlashIfNotExist($file);
+        if (!in_array($file, $this->lines)) {
+            $this->lines[] = $file;
+        }
+        $this->hasChanges = true;
+    }
+    
+    /**
+     * @param array $files
+     */
+    public function addMultipleEntries(array $files)
+    {
+        foreach ($files as $file) {
+            $this->addEntry($file);
+        }
+    }
+    
+    /**
+     * @param string $file
+     */
+    public function removeEntry($file)
+    {
+        $file = $this->prependSlashIfNotExist($file);
+        $key = array_search($file, $this->lines);
+        if (false !== $key) {
+            unset($this->lines[$key]);
+            $this->hasChanges = true;
+            // renumber array
+            $this->lines = array_values($this->lines);
+        }
+    }
+    
+    /**
+     * @param array $files
+     */
+    public function removeMultipleEntries(array $files)
+    {
+        foreach ($files as $file) {
+            $this->removeEntry($file);
+        }
+    }
+    
+    /**
+     * @return array
+     */
+    public function getEntries()
+    {
+        return $this->lines;
+    }
+    
+    /**
+     * Write the file
+     */
+    
+    public function write()
+    {
+        if ($this->hasChanges) {
```

Comment:
You have to check here if the file exists or not and create it before if doesn't exist